taca/nanopore/ONT_run_classes.py

```diff
@@ -338,6 +350,113 @@ def copy_html_report(self):
+
+        # Get sequencing summary file
+        glob_summary = glob.glob(f"{self.run_abspath}/sequencing_summary*.txt")
+        assert len(glob_summary) == 1, f"Found {len(glob_summary)} summary files"
+        summary = glob_summary[0]
+
+        # Determine the format of the raw sequencing data, sorted by preference
+        raw_data_dir_options = [
+            "pod5_pass",
+            "pod5",
+            "fast5_pass",
+            "fast5",
+        ]
+        raw_data_dir = None
+        for raw_data_dir_option in raw_data_dir_options:
+            if os.path.exists(f"{self.run_abspath}/{raw_data_dir_option}"):
+                raw_data_dir = raw_data_dir_option
+                break
+        if raw_data_dir is None:
+            raise AssertionError(f"No seq data found in {self.run_abspath}")
+        raw_data_format = "pod5" if "pod5" in raw_data_dir else "fast5"
+
+        # Load samplesheet, if any
+        ss_glob = glob.glob(f"{self.run_abspath}/sample_sheet*.csv")
+        if len(ss_glob) == 0:
+            samplesheet = None
+        elif len(ss_glob) > 1:
+            # If multiple samplesheet, use latest one
+            samplesheet = ss_glob.sort()[-1]
+        else:
+            samplesheet = ss_glob[0]
+
+        # Run has barcode subdirs
+        if len(glob.glob(f"{self.run_abspath}/fastq_pass/barcode*")) > 0:
+            barcode_dirs = True
+        else:
+            barcode_dirs = False
+
+        # Determine barcodes
+        if samplesheet:
+            ss_df = pd.read_csv(samplesheet)
+            if "barcode" in ss_df.columns:
+                ss_barcodes = list(ss_df["barcode"].unique())
+                ss_barcodes.sort()
+                barcode_nums = [int(bc[-2:]) for bc in ss_barcodes]
+                # If barcodes are numbered sequentially, write arg as range
+                if barcode_nums == list(range(1, len(barcode_nums) + 1)):
+                    barcodes_arg = f"{ss_barcodes[0]}:{ss_barcodes[-1]}"
+                else:
+                    barcodes_arg = ":".join(ss_barcodes)
+            else:
+                ss_barcodes = None
+
+        command_args = {
+            "--sequencing-summary-source": summary,
+            f"--{raw_data_format}-source": f"{self.run_abspath}/{raw_data_dir}",
+            "--output-directory": self.run_abspath,
+            "--report-name": "toulligqc_report",
+        }
+        if barcode_dirs:
+            command_args["--barcoding"] = ""
+            if samplesheet and ss_barcodes:
+                command_args["--samplesheet"] = samplesheet
+                command_args["--barcodes"] = barcodes_arg
+
+        # Build command list
+        command_list = [self.toulligqc_executable]
+        for k, v in command_args.items():
+            command_list.append(k)
+            if v:
+                command_list.append(v)
+
+        # Run the command
+        # Small enough to wait for, should be done in 1-5 minutes
+        process = subprocess.run(command_list)
```

Comment:
Does it create any files besides the report? Do we need to clean it up sometimes?

Reply:
The tools should create a dir containing the .html report as well as subdirs `data` and `images`. I don't currently see why we'd need cleanup.

Reply:
The tools should create a dir containing the .html report as well as subdirs `data` and `images`. I don't currently see why we'd need cleanup.